src/ui/Menus/PostponeMenu.ts

```diff
@@ -38,11 +38,23 @@ export class PostponeMenu extends TaskEditingMenu {
             itemNamingFunction: NamingFunction,
             postponingFunction: PostponingFunction,
         ) => {
+            let postponeDateEqualsTaskFieldDate = false;
+            const dateFieldToPostpone = getDateFieldToPostpone(task);
+            if (dateFieldToPostpone) {
+                const { postponedDate } = postponingFunction(task, dateFieldToPostpone!, timeUnit, amount);
```

Comment:
sonarcloud reports a warning on the `dateFieldToPostpone!`:

> This assertion is unnecessary since it does not change the type of the expression.

Reply:
Oh, thanks, I missed that